examples/utils/migration-tools/README.md

```diff
@@ -2,7 +2,93 @@
 
 This package contains tools for migrating data from one version to another, used by Fluid examples. They are not currently intended for use in production scenarios.
 
-See [GitHub](https://github.com/microsoft/FluidFramework) for more details on the Fluid Framework and packages within.
+Use of the migration tools imposes several requirements on the container code and application, detailed here.
+
+## Implementing `IMigratableModel`
+
+Your data model must implement `IMigratableModel` to be migrated using the migration tools.
+
+This includes:
+1. A `version` string to identify the model version.
+1. Methods to export and import data, and to detect if the model supports a given data format:
+	1. `importData: (initialData: ImportType) => Promise<void>`
+	1. `exportData: () => Promise<ExportType>`
+	1. `supportsDataFormat: (initialData: unknown) => initialData is ImportType`
+1. A `dispose` method to clean up the container - most likely calling `IContainer.dispose`.
+
+## Implementing the composite runtime pattern
+
+See documentation for the composite runtime pattern [here](./src/compositeRuntime/README.md).
+
+The migration tools expect to find an `IMigratableModel` by accessing and calling a `getModel()` function provided on the `entryPoint`.  They also expect to find an `IMigrationTool` by accessing a `migrationTool` member of the `entryPoint`.  These requirements are most easily satisfied by using the composite runtime pattern.
+
+`getModel()` is a function that takes an `IContainer` to aid in producing the `IMigratableModel`.  This is because the contract of `IMigratableModel` likely requires functionality from `IContainer` (especially `IContainer.dispose()`).
+
+### Defining the entry point piece
+
+```ts
+const rootDatastoreAlias = "my-root-datastore";
+
+export const getModelEntryPointPiece: IEntryPointPiece = {
+	name: "getModel",
+	registryEntries: [MyRootDatastoreFactory.registryEntry],
+	onCreate: async (runtime: IContainerRuntime): Promise<void> => {
+		const rootDatastore = await runtime.createDataStore(MyRootDatastoreFactory.type);
+		await rootDatastore.trySetAlias(rootDatastoreAlias);
+	},
+	onLoad: async (runtime: IContainerRuntime): Promise<void> => {},
+	createPiece: async (runtime: IContainerRuntime): Promise<MyRootDatastore> => {
```

Comment:
The return type should be `Promise<(container: IContainer) => Promise<MigratableAppModel>>` or approximately that, correct? (not `Promise<MyRootDatastore>`)

Reply:
Yes, thank you - I was flipping back and forth between this README and the compositeRuntime/README.md and I apparently forgot to fixup a few things.  Fixed in next.